backend/core/src/listings/listings.service.ts

```diff
@@ -212,4 +214,22 @@ export class ListingsService {
     }
     return listing
   }
+
+  @Interval(1000 * 60 * 60)
+  public async changeOverdueListingsStatusCron() {
+    const listings = await this.listingRepository
+      .createQueryBuilder()
+      .select(["id", "applicationDueDate", "status"])
+      .getMany()
+    let shouldSave = false
+    for (const listing of listings) {
+      if (listing.applicationDueDate && listing.applicationDueDate < new Date()) {
+        listing.status = ListingStatus.closed
+        shouldSave = true
+      }
+    }
+    if (shouldSave) {
+      await this.listingRepository.save(listings)
+    }
+  }
```

Comment:
Can you update this to query for listings where the application due date is in the past and the status is active and only update those.

Reply:
And with that update, update the tests to include active, pending and closed listings with past and present due dates and ensure only the applicable ones are updated.